net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -155,6 +219,175 @@ public static RuleMetadata getRuleMetadata(String name) {
     return metadata;
   }
 
+  /**
+   * Create metadata for modules not present in the previously eclipse provided metadata.
+   * Work in progress.
+   *
+   * @param moduleDetails module details fetched from checkstyle metadata
+   * @return ruleMetadata for the module
+   */
+  public static RuleMetadata createRuleMetadata(ModuleDetails moduleDetails) {
+    final String[] packageTokens = moduleDetails.getFullQualifiedName().split("\\.");
+    RuleGroupMetadata group;
+    String modulePackageNameString = moduleDetails.getFullQualifiedName();
+    // standard checkstyle module
+    if (modulePackageNameString.contains("com.puppycrawl.tools.checkstyle")) {
+      group = getRuleGroupMetadata(sPackageToGroupNameMap
+              .get(packageTokens[packageTokens.length - 2]));
+    }
+    // third party extension modules
+    else {
+      String lookupKey = findCorrectLookupKey(modulePackageNameString);
+      String ruleGroupNameString = sThirdPartyRuleGroupMap.get(lookupKey).get("name");
+      group = getRuleGroupMetadata(ruleGroupNameString);
+      // if the group of the new check hasn't been formed yet
+      // and put into the sRuleGroupMetadata map
+      if (group == null) {
+        group = new RuleGroupMetadata(ruleGroupNameString,
+                sThirdPartyRuleGroupMap.get(lookupKey).get("description"), false,
+                Integer.parseInt(sThirdPartyRuleGroupMap.get(lookupKey).get("priority")));
```

Comment:
move `sThirdPartyRuleGroupMap.get(lookupKey).get("description")`
and `Integer.parseInt(sThirdPartyRuleGroupMap.get(lookupKey).get("priority"))` to local variables to easy code reading.

Reply:
Done